tools/souffle_converter.py

```diff
@@ -1046,14 +1047,87 @@ def convert_terms(self, terms):
                 result.append(term)
         return result + tail
 
+    def flatten(self, tail, listbody, bool_dis):
+        for x in tail.children:
+            if (x.symbol.name == "Term"):
+                if (getOptField(x, "Literal") != None) or (getOptField(x, "!") != None):
+                    listbody.append(x)
```

Comment:
if "continue" would work here I think it would make the code easier to understand - and perhaps faster too.